### ISSUE TITLE
use pkg-config to find parameters to link against libpq

### ISSUE THREAD

**Issue by darix** [jeremyevans/sequel_pg#31]
pg_config was never really meant to be used for programs using libpq.
But for lack of better alternatives everyone started using it in the
past. There is a better alternative now pkg-config libpq.

```
Index: sequel_pg-1.12.0/ext/sequel_pg/extconf.rb
===================================================================
--- sequel_pg-1.12.0.orig/ext/sequel_pg/extconf.rb
+++ sequel_pg-1.12.0/ext/sequel_pg/extconf.rb
@@ -1,8 +1,10 @@
 require 'mkmf'
 $CFLAGS << " -O0 -g" if ENV['DEBUG']
 $CFLAGS << " -Wall " unless RUBY_PLATFORM =~ /solaris/
+if !pkg_config('libpq')
 dir_config('pg', ENV["POSTGRES_INCLUDE"] || (IO.popen("pg_config --includedir").readline.chomp rescue nil),                                                                                  
                  ENV["POSTGRES_LIB"]     || (IO.popen("pg_config --libdir").readline.chomp rescue nil))                                                                                      
+end

 if (have_library('pq') || have_library('libpq') || have_library('ms/libpq')) && have_header('libpq-fe.h')                                                                                    
   have_func 'PQsetSingleRowMode'
```

**Comment by jeremyevans** [comment]
I'm OK with the general idea of using `pkg_config`.  We should probably only use `pkg_config` if `ENV["POSTGRES_INCLUDE"]` and `ENV["POSTGRES_LIB"]` are not specified, though, otherwise you break existing code where users are trying to link against custom PostgreSQL installations.

**Comment by darix** [comment]
They can still do that via export PKG_CONFIG_PATH=...

**Comment by jeremyevans** [comment]
That's all well and good.  However, if they have existing infrastructure that uses POSTGRES_INCLUDE and POSTGRES_LIB, I don't want to break that.

**Comment by darix** [comment]
That makes the whole conditional a bit ugly but lets see...

**Comment by darix** [comment]
so something like this?

```
Index: sequel_pg-1.12.0/ext/sequel_pg/extconf.rb
===================================================================
--- sequel_pg-1.12.0.orig/ext/sequel_pg/extconf.rb
+++ sequel_pg-1.12.0/ext/sequel_pg/extconf.rb
@@ -1,8 +1,14 @@
 require 'mkmf'
 $CFLAGS << " -O0 -g" if ENV['DEBUG']
 $CFLAGS << " -Wall " unless RUBY_PLATFORM =~ /solaris/
-dir_config('pg', ENV["POSTGRES_INCLUDE"] || (IO.popen("pg_config --includedir").readline.chomp rescue nil),
-                 ENV["POSTGRES_LIB"]     || (IO.popen("pg_config --libdir").readline.chomp rescue nil))
+if !ENV["POSTGRES_INCLUDE"].nil? or !ENV["POSTGRES_LIB"].nil?
+  dir_config('pg', ENV["POSTGRES_INCLUDE"], ENV["POSTGRES_LIB"])
+else
+  if !pkg_config('libpq')
+    dir_config('pg', (IO.popen("pg_config --includedir").readline.chomp rescue nil),
+                     (IO.popen("pg_config --libdir").readline.chomp rescue nil))
+  end
+end
 
 if (have_library('pq') || have_library('libpq') || have_library('ms/libpq')) && have_header('libpq-fe.h')
   have_func 'PQsetSingleRowMode'
```

**Comment by darix** [comment]
we could also skip the pg_config fallback and error out then.

**Comment by jeremyevans** [comment]
I was thinking something along the lines of:

```ruby
if ENV["POSTGRES_INCLUDE"] || ENV["POSTGRES_LIB"] || !pkg_config('libpq')
  dir_config(...)
end
```

**Comment by jeremyevans** [comment]
Fixed by 529c210acc0c9e8586d8eff1788c72187ad884a3.
